src/inference/Inference.jl

```diff
@@ -249,15 +249,11 @@ function _params_to_array(ts::Vector)
     # Extract the parameter names and values from each transition.
     dicts = map(ts) do t
         nms, vs = flatten_namedtuple(getparams(t))
-        for nm in nms
-            if !(nm in names)
-                push!(names, nm)
-            end
-        end
+        append!(names, nms)
         # Convert the names and values to a single dictionary.
         return Dict(nms[j] => vs[j] for j in 1:length(vs))
     end
-    # names = collect(names_set)
+    unique!(names)
```

Comment:
Probably we should just use a `Set` (actually, I thought we operate on `Set`s but maybe this was lost in some refactoring).

Reply:
I think I changed it away from using a `Set` because the order was getting messed up, though I cannot recall the exact reason.

Reply:
It may not be relevant anymore, though.

Reply:
I wonder if an [`OrderedSet`](https://juliacollections.github.io/DataStructures.jl/v0.8/ordered_containers.html) would be faster.

Reply:
`OrderedSet` seems perfect 👍